### ISSUE TITLE
Bug in block_cohortspecifichtml_pluginfile() function

### ISSUE THREAD

**Issue by iandavidwild** [moodleuulm/moodle-block_cohortspecifichtml#12]
I think you may have an operator precedence issue in the function _block_cohortspecifichtml_pluginfile()_ in _lib.php_. Instead of setting variable _$file_ to a file object, lines 78-80 are setting variable _$file_ to **true**:

    if (!$file = $fs->get_file($context->id, 'block_cohortspecifichtml', 'content',
            0, $filepath, $filename) || $file->is_directory()) {
        send_file_not_found();
    }

To address this issue, we have split the code as follows:

    $file = $fs->get_file($context->id, 'block_cohortspecifichtml', 'content',
            0, $filepath, $filename);
    if ( ($file === false) || ($file->is_directory() === true)) {
        send_file_not_found();
    }

Hope this helps.

**Comment by abias** [comment]
Oh my.

This plugin is based on block_html from Moodle core and adopts many pieces of codes from there. 
This particular piece of code is also taken from block_html.

Now this regression sneaked in recently when I tried to make codechecker happy again in https://github.com/moodle-an-hochschulen/moodle-block_cohortspecifichtml/commit/229af425e8aebef4298d4cdbf7ea6cbe9106c91a. This was done because of a codechecker complaint about `or` operators which are now deprecated. I changed this to `||` but I missed the fact that `or` is weaker as `=`, but `||` is not.

I have fixed this now based on your proposal and have published new plugin versions.

Cheers,
Alex
